opnode/l1/downloader.go

```diff
@@ -11,24 +12,34 @@ import (
 	"github.com/ethereum/go-ethereum/core/types"
 )
 
+// TODO: Make these configurable and part of a configuration object
 const MaxConcurrentFetchesPerCall = 10
 const MaxReceiptRetry = 3
+const MaxBlocksInL1Range = uint64(100)
 
-type EthClient interface {
+type DownloaderClient interface {
 	BlockByHash(context.Context, common.Hash) (*types.Block, error)
+	BlockByNumber(context.Context, *big.Int) (*types.Block, error)
 	TransactionReceipt(context.Context, common.Hash) (*types.Receipt, error)
 }
 
+// Block
+// Receipts
+// Transactions (from range of blocks)
+
 type Downloader struct {
-	client EthClient
+	client DownloaderClient
```

Comment:
Why do we need to wrap this in a struct, do we plan to add things in there in the future?

Reply:
It does enable us to add things in the future (like a cache), but the primary thing that it enables is exposing different function signatures.